docs/modules/ROOT/pages/how-tos/exoscale/install.adoc

```diff
@@ -101,66 +110,138 @@ openshift-install --dir ${CLUSTER_ID} \
+exo vm template register "rhcos-${RHCOS_VERSION}" \
   --checksum $(md5sum rhcos-${RHCOS_VERSION}.qcow2 | awk '{ print $1 }') \
   --boot-mode uefi \
   --disable-password \
   --username core \
   --description "Red Hat Enterprise Linux CoreOS (RHCOS) ${RHCOS_VERSION}" \
-  --name "rhcos-${RHCOS_VERSION}" \
-  --url "https://${AWS_S3_ENDPOINT}/${CLUSTER_ID}-bootstrap-ignition/rhcos-${RHCOS_VERSION}.qcow2"
+  --url "https://${AWS_S3_ENDPOINT}/${CLUSTER_ID}-bootstrap/rhcos-${RHCOS_VERSION}.qcow2"
+
+exo sos remove "${CLUSTER_ID}-bootstrap" rhcos-${RHCOS_VERSION}.qcow2
 
 export TF_VAR_rhcos_template="rhcos-${RHCOS_VERSION}"
 ----
 
+. Upload Ursula Debian package to bootstrap bucket
++
+[source,console]
+----
+exo sos upload "${CLUSTER_ID}-bootstrap" /path/to/ursula.deb
+exo sos acl add "${CLUSTER_ID}-bootstrap" ursula.deb --public-read
+----
+
 . Initialize Terraform
 +
+[NOTE]
+====
+Exoscale doesn't allow uploading the same public key multiple times in a single organisation.
+If the public key in `$SSH_PUBLIC_KEY` already exists in the organisation, we instruct Terraform to use the existing keypair instead of trying to upload the public key under a new name.
+====
++
 [source,console]
 ----
-terraform init \
-  -backend-config "bucket=${CLUSTER_ID}-tf-state"
+cat > backend.tf <<EOF
+terraform {
+  backend "s3" {
+    key                         = "cluster.tfstate"
+    region                      = "us-east-1" # Ignored
+    bucket                      = "${CLUSTER_ID}-tf-state"
+    skip_credentials_validation = true
+    skip_metadata_api_check     = true
+  }
+}
+EOF
+terraform init
 
 export TF_VAR_cluster_id=$CLUSTER_ID
-export TF_VAR_ssh_key="$(cat ~/.ssh/id_rsa.pub)"
+export TF_VAR_base_domain=${BASE_DOMAIN}
+export TF_VAR_ssh_key="$(cut -d' ' -f1,2 <$SSH_PUBLIC_KEY)"
+export TF_VAR_existing_keypair=$(exo sshkey list -Ojson | \
+        jq -r ".[] | select(.fingerprint == \"${SSH_PUBKEY_FINGERPRINT}\") | .name")
 ----
 
-. Set up DNS NS records on parent zone (see Terraform output `ns_records`)
+. Create Exoscale domain
 +
 [source,console]
 ----
-terraform apply \
-  -target exoscale_domain_record.exo_nameservers
+terraform apply -target data.exoscale_domain_record.exo_nameservers
 ----
 
-. Bootstrap cluster
+. Set up DNS NS records on parent zone using the data from the Terraform output variable `ns_records`
+
+. Deploy LBs and bootstrap node
++
+[NOTE]
+====
+We initially provision VMs in "Stopped" state to ensure the private network interface is attached when the VM powers up.
+We provision the VMs in this two-step process to avoid VMs getting stuck in a state where they're unable to reach the machine config server when deploying the cluster into a private network.
+====
++
+[NOTE]
+====
+We currently need another call to `terraform apply` after changing the VM state from `Stopped` to `Running`.
+This is necessary because the Exoscale Terraform provider doesn't correctly signal that a VMs IP might change during a state change.
```

Comment:
```suggestion
This is necessary because the Exoscale Terraform provider doesn't correctly signal that a VM's IP might change during a state change.
```